tests/unit_tests/snapshot_test.py

```diff
@@ -0,0 +1,387 @@
+                 "using_data_variable \"sysctl_sched_latency\"",
+                 "kernel.sched_latency_ns")
+
+    assert "sched_debug_header" in snap.fun_groups[
+        "kernel.sched_latency_ns"].functions
+    fun_desc = snap.fun_groups["kernel.sched_latency_ns"].functions[
+        "sched_debug_header"]
+    assert fun_desc.mod is mod
+    assert fun_desc.glob_var == "sysctl_sched_latency"
+    assert fun_desc.tag == "using_data_variable \"sysctl_sched_latency\""
+
+
+def test_get_modules():
+    """
+    Test getting all modules in the snapshot function lists.
+    Check if the snapshot returns a list of all modules of all groups in case
+    that multiple groups are present.
+    """
+    snap = Snapshot(fun_kind="sysctl")
+
+    snap.add_fun_group("kernel.sched_latency_ns")
+    snap.add_fun_group("kernel.timer_migration")
+    snap.add_fun("sched_proc_update_handler",
+                 LlvmKernelModule("kernel/sched/fair.ll"), None,
+                 "proc_handler", "kernel.sched_latency_ns")
+    snap.add_fun("proc_dointvec_minmax", LlvmKernelModule("kernel/sysctl.ll"),
+                 None,
+                 "proc_handler", "kernel.timer_migration")
+
+    modules = snap.modules()
+    assert len(modules) == 2
+    assert set([m.llvm for m in modules]) == {"kernel/sched/fair.ll",
+                                              "kernel/sysctl.ll"}
+
+
+def test_get_by_name_functions():
+    """Get the module of inserted function by its name."""
+    snap = Snapshot()
+
+    snap.add_none_group()
+    mod_buff = LlvmKernelModule("net/core/skbuff.ll")
+    mod_alloc = LlvmKernelModule("mm/page_alloc.ll")
+    snap.add_fun("___pskb_trim", mod_buff)
+    snap.add_fun("__alloc_pages_nodemask", mod_alloc)
+
+    fun = snap.get_by_name("___pskb_trim")
+    assert fun.mod is mod_buff
+    fun = snap.get_by_name("__alloc_pages_nodemask")
+    assert fun.mod is mod_alloc
+
+
+def test_get_by_name_sysctls():
+    """Get the module of inserted function by its name and sysctl group."""
+    snap = Snapshot(fun_kind="sysctl")
+
+    snap.add_fun_group("kernel.sched_latency_ns")
+    snap.add_fun_group("kernel.timer_migration")
+    mod_fair = LlvmKernelModule(
+        "snapshots-sysctl/linux-3.10.0-957.el7/kernel/sched/fair.ll")
+    mod_sysctl = LlvmKernelModule(
+        "snapshots-sysctl/linux-3.10.0-957.el7/kernel/sysctl.ll")
+    snap.add_fun("sched_proc_update_handler", mod_fair, None, "proc handler",
+                 "kernel.sched_latency_ns")
+    snap.add_fun("proc_dointvec_minmax", mod_sysctl, None, "proc handler",
+                 "kernel.timer_migration")
+
+    fun = snap.get_by_name("sched_proc_update_handler",
+                           "kernel.sched_latency_ns")
+    assert fun.mod is mod_fair
+    fun = snap.get_by_name("proc_dointvec_minmax", "kernel.sched_latency_ns")
+    assert fun is None
+    fun = snap.get_by_name("sched_proc_update_handler",
+                           "kernel.timer_migration")
+    assert fun is None
```

Comment:
I think that this is a duplicate test of the above (getting function for a wrong group), could be removed. Maybe even add a comment why `fun` is expected to be `None`.